src/Sulu/Bundle/SnippetBundle/EventListener/DomainEventSubscriber.php

```diff
@@ -0,0 +1,348 @@
+<?php
+
+/*
+ * This file is part of Sulu.
+ *
+ * (c) Sulu GmbH
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Sulu\Bundle\SnippetBundle\EventListener;
+
+use PHPCR\NodeInterface;
+use Sulu\Bundle\DocumentManagerBundle\Collector\DocumentDomainEventCollectorInterface;
+use Sulu\Bundle\SnippetBundle\Document\SnippetDocument;
+use Sulu\Bundle\SnippetBundle\Domain\Event\SnippetCopiedEvent;
+use Sulu\Bundle\SnippetBundle\Domain\Event\SnippetCreatedEvent;
+use Sulu\Bundle\SnippetBundle\Domain\Event\SnippetModifiedEvent;
+use Sulu\Bundle\SnippetBundle\Domain\Event\SnippetRemovedEvent;
+use Sulu\Bundle\SnippetBundle\Domain\Event\SnippetTranslationAddedEvent;
+use Sulu\Bundle\SnippetBundle\Domain\Event\SnippetTranslationCopiedEvent;
+use Sulu\Bundle\SnippetBundle\Domain\Event\SnippetTranslationRemovedEvent;
+use Sulu\Component\Content\Document\Extension\ExtensionContainer;
+use Sulu\Component\Content\Document\Subscriber\SecuritySubscriber;
+use Sulu\Component\DocumentManager\DocumentManagerInterface;
+use Sulu\Component\DocumentManager\Event\CopyEvent;
+use Sulu\Component\DocumentManager\Event\CopyLocaleEvent;
+use Sulu\Component\DocumentManager\Event\FlushEvent;
+use Sulu\Component\DocumentManager\Event\PersistEvent;
+use Sulu\Component\DocumentManager\Event\RemoveEvent;
+use Sulu\Component\DocumentManager\Event\RemoveLocaleEvent;
+use Sulu\Component\DocumentManager\Events;
+use Sulu\Component\DocumentManager\PropertyEncoder;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Webmozart\Assert\Assert;
+
+/**
+ * @internal
+ */
+class DomainEventSubscriber implements EventSubscriberInterface
```

Comment:
would you mind to add some checks in the `SnippetControllerTest` like here?
https://github.com/sulu/sulu/pull/5988/files#diff-e053f115525b3f9f06c8c92b827a46c8b284508dd1f7fb45170642e1cd689171

would be nice to have a functional test case for phpcr events 🙂